README.md

```diff
@@ -6,108 +6,206 @@
 > This function relies on functionalities only available in Crossplane 1.15 and
 > later. It will not work with earlier versions.
 
+> [!IMPORTANT]  
+> This function is meant to replace native Composition Environment
+> (`--enable-environment-configs`), see
+> [below](#migration-from-native-composition-environment) for more details.
+
 This [composition function][docs-functions] allows you to request
 `EnvironmentConfigs`, merge them in the requested order and return inject the
 computed `environment` into the `Context` at a well-known key,
 `apiextensions.crossplane.io/environment`, so that other functions such as
 [function-patch-and-transform] can access it.
 
-The following paths from native `Resource` Compositions can be just moved to the
-input of this function:
+## Examples
```

Comment:
This is now kinda duplicating the `## Using this function` section

Reply:
merged and moved that here.